tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp

```diff
@@ -117,49 +117,55 @@ bool is_parametrized_type(T val) {
 
 struct UnaryWithParam {
     UnaryOpType op_type;
-    std::optional<float> param = std::nullopt;
+    std::vector<float> params;
+
+    UnaryWithParam(UnaryOpType op_type, const std::vector<float>& params) : op_type{op_type}, params{params} {}
+    UnaryWithParam(UnaryOpType op_type, float param) : op_type{op_type}, params{param} {}
+    UnaryWithParam(UnaryOpType op_type) : op_type{op_type} {}
+
+    bool has_parameter() const { return params.size() > 0; }
 
     static constexpr auto attribute_names = std::make_tuple("op_type", "param");
-    const auto attribute_values() const { return std::make_tuple(std::cref(this->op_type), std::cref(this->param)); }
+    const auto attribute_values() const { return std::make_tuple(std::cref(this->op_type), std::cref(this->params)); }
```

Comment:
Please advise on what should be done here.

Reply:
Can the op be made dynamic on these `params`? ie. Can the args in these params be used as runtime args? If so, we can just remove `params` from attributes.

Reply:
What are the attributes used for? Is it caching?

Reply:
These are all injected as defines, so can't really be used as rt args right now